### PR TITLE
Adds DumpNodes method to Kademlia Inspector

### PR DESCRIPTION
**Pull request by dylanlott**

### CODE REVIEW

File: pkg/kademlia/routing.go [hunk 1]
```diff
@@ -147,6 +147,39 @@ func (rt *RoutingTable) GetBucketIds() (storage.Keys, error) {
 	return kbuckets, nil
 }
 
+// DumpNodes loops returns all nodes in the nodeBucketDB and marshals them to &pb.Nodes
+func (rt *RoutingTable) DumpNodes() ([]*pb.Node, error) {
```
Comment (wthorp) [inline pkg/kademlia/routing.go]: Jen is working on replacing all the calls to List() w/ iterators [https://github.com/storj/storj/pull/1318].  I'd use routing.go's iterate() https://github.com/storj/storj/blob/36e4d8c1c28594cb9b30950db227cc7fd025886e/pkg/kademlia/routing.go#L262 here

File: pkg/kademlia/routing.go [hunk 2]
```diff
@@ -149,6 +149,19 @@ func (rt *RoutingTable) GetBucketIds() (storage.Keys, error) {
 	return kbuckets, nil
 }
 
+// DumpNodes iterates through all nodes in the nodeBucketDB and marshals them to &pb.Nodes, then returns them
+func (rt *RoutingTable) DumpNodes() ([]*pb.Node, error) {
+	var nodes []*pb.Node
+	err := rt.iterate(storj.NodeID{}, func(newID storj.NodeID, protoNode []byte) error {
+		newNode := pb.Node{}
+		err := proto.Unmarshal(protoNode, &newNode)
+		nodes = append(nodes, &newNode)
```
Comment (mniewrzal) [inline pkg/kademlia/routing.go]: Should we append when err != nil?

File: pkg/kademlia/routing.go [hunk 3]
```diff
@@ -149,6 +149,19 @@ func (rt *RoutingTable) GetBucketIds() (storage.Keys, error) {
...
+		return err
+	})
+
+	return nodes, Error.Wrap(err)
```
Comment (mniewrzal) [inline pkg/kademlia/routing.go]: I think its safer to return nil array when an error occurs.

File: pkg/overlay/inspector.go [hunk 4]
```diff
@@ -30,3 +31,8 @@ func (srv *Inspector) CountNodes(ctx context.Context, req *pb.CountNodesRequest)
 		Count: int64(len(overlayKeys)),
 	}, nil
 }
+
+// DumpNodes returns all of the nodes in the overlay cachea
```
Comment (mniewrzal) [inline pkg/overlay/inspector.go]: typo - cachea

File: pkg/kademlia/routing.go (hunk 2 above)
Comment (dylanlott) [reply pkg/kademlia/routing.go]: Good point. I'll rework this to use ErrorGroup handling.

File: pkg/kademlia/routing.go (hunk 3 above)
Comment (dylanlott) [reply pkg/kademlia/routing.go]: Same as above, using ErrorGroup here for this.

File: pkg/overlay/inspector.go (hunk 4 above)
Comment (dylanlott) [reply pkg/overlay/inspector.go]: Resolved

**Pull request merged**